File: 88.go
func merge(nums1 []int, m int, nums2 []int, n int)  {
	sorted := []int{}

	p1 := 0
	p2 := 0

	for p1 < m || p2 < n {
		if p1 >= m {
			sorted = append(sorted, nums2[p2:]...)
			break
		}

		if p2 >= n {
			sorted = append(sorted, nums1[p1:]...)
			break
		}

		if nums1[p1] < nums2[p2] {
			sorted = append(sorted, nums1[p1])
			p1++
		} else {
			sorted = append(sorted, nums2[p2])
			p2++
		}
	}

	copy(nums1, sorted)
}

File: 88.ts
/**
 Do not return anything, modify nums1 in-place instead.
 */
function merge(nums1: number[], m: number, nums2: number[], n: number): void {
	const sorted: number[] = [];

	let p1 = 0;
	let p2 = 0;

	while (p1 < m || p2 < n) {
		if (p1 >= m) {
			sorted.push(...nums2.slice(p2))
			break
		}
		if (p2 >= n) {
			sorted.push(...nums1.slice(p1))
			break
		}

		if (nums1[p1] > nums2[p2]) {
			sorted.push(nums2[p2])
			p2++
		} else {
			sorted.push(nums1[p1])
			p1++
		}
	}

	for (let i = 0; i < nums1.length; i++) {
		nums1[i] = sorted[i]
	}
};